modules/nw-ovn-kubernetes-migration.adoc

```diff
@@ -66,7 +113,30 @@ do
 done
 ----
 
-. After the nodes in your cluster have rebooted, confirm that the migration succeeded:
+. After the nodes in your cluster have rebooted, start all of the machine configuration pools. As the MCO updates machines in each config pool, by default it updates a single machine per pool at a time, so the time that the migration requires to complete grows with the size of the cluster.
++
+[source,terminal]
+----
+$ oc patch MachineConfigPool master --type='merge' --patch \
+  '{ "spec": { "paused": false } }'
+----
++
+[source,terminal]
+----
+$ oc patch MachineConfigPool worker --type='merge' --patch \
+  '{ "spec": { "paused": false } }'
+----
+
+. To confirm the status of the new machine configuration on the hosts, enter the following command.
++
+[source,terminal]
+----
+$ oc describe node | egrep "hostname|machineconfig"
+----
++
+When the new configuration is applied, the value of `machineconfiguration.openshift.io/state` field is `Done`.
```

Comment:
Users shall check whether the machineconfiguration.openshift.io/currentConfig of all the nodes from one machine config pool is the same and is the name of the latest MachineConfig.

Reply:
@pliurh is this in addition to `machineconfiguration.openshift.io/state`?

How can someone do this? Is there a set of commands I can include for this verification step?

Thanks.

Reply:
Here comes the output example of `oc describe node | egrep "hostname|machineconfig"` for one node.

```
                    kubernetes.io/hostname=master-0
                    machineconfiguration.openshift.io/currentConfig: rendered-master-c53e221d9d24e1c8bb6ee89dd3d8ad7b
                    machineconfiguration.openshift.io/desiredConfig: rendered-master-c53e221d9d24e1c8bb6ee89dd3d8ad7b
                    machineconfiguration.openshift.io/reason:
                    machineconfiguration.openshift.io/state: Done
```

When the `state` is done, the `currentConfig` shall equal to the `desiredConfig`. The name of the MachineConfig can be found by `oc get machineconfig`. When we trigger the migration, a new MachineConfig will be rendered by MCO for each MachineConfigPool. Users have to make sure that the newly rendered MachineConfig has been applied to the node successfully by comparing the name of `currentConfig` and the output of `oc get machineconfig`.

```
[root@nfvpe-21 ~]# oc get mc
...
rendered-master-c53e221d9d24e1c8bb6ee89dd3d8ad7b    6c8bd9b2915c44e03ec10e02e59fedb07e3807cb   3.1.0             6d23h
...
```

Reply:
@pliurh, the challenge with MCO is that the `rendered-master-...` hash is mostly meaningless. If there are a dozen of these in the list, how does someone know that `c53e221d9d24e1c8bb6ee89dd3d8ad7b` is what must be in `machineconfiguration.openshift.io/currentConfig`?

If it is the most recently created resource, that might be a good guess. But it isn't definitive.

Is it possible to sort `oc get mc` by creation time?

Does `oc get mc rendered-master-c53e221d9d24e1c8bb6ee89dd3d8ad7b -o yaml` provide any useful information? Is it possible to see if `rendered-master-c53e221d9d24e1c8bb6ee89dd3d8ad7b` is correct some other way than listing all machineconfig resources?

Reply:
I always use the creation time as an indicator.  However, `oc get mc rendered-master-c53e221d9d24e1c8bb6ee89dd3d8ad7b -o yaml` can help as well. In the ignition config of the machine config CR, user can check the systemd unit of `ovs-configuration`. If it's for ovn-kubernetes, you can see `ExecStart=/usr/local/bin/configure-ovs.sh OVNKubernetes`; for openshift-sdn, you can see `ExecStart=/usr/local/bin/configure-ovs.sh OpenShiftSDN`

Reply:
@pliurh, do you have any example command output for:

`oc get mc rendered-master-c53e221d9d24e1c8bb6ee89dd3d8ad7b -o yaml`

That I can include in the document?

Thanks!